lib/miq_automation_engine/engine/miq_ae_service.rb

```diff
@@ -237,6 +237,42 @@ def execute(m, *args)
       raise MiqAeException::MethodNotFound, err.message
     end
 
+    def notification_subject(values_hash)
+      subject = values_hash[:subject] || @workspace.ae_user
+      subject = ar_object(subject) || subject
+      raise ArgumentError, "Subject must be a valid Active Record object" unless subject.kind_of?(ActiveRecord::Base)
+      subject
+    end
+
+    def ar_object(svc_obj)
+      if svc_obj.kind_of?(MiqAeMethodService::MiqAeServiceModelBase)
+        svc_obj.instance_variable_get('@object')
+      end
+    end
+
+    def notification_type(values_hash)
+      if values_hash[:type].present?
```

Comment:
@tinaafitz 
this could be
```
type = if values_hash[:type].present?
               values_hash[:type].to_sym
            else
               default_notification_type(values_hash)
            end